README.md

```diff
@@ -145,7 +174,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+script.py  
```

Comment:
# 9

First, let me say that I like your coding style: very clear to read.

I have two problems with your submission, though. First, your expressed the transitions in code, where the problem asked you to code the table from the previous question (the assumption being that you'd create a data structure).

Second, rule S1 has an arbitrary `range(index, index + 3)` in it, which means it doesn't actually match the grammar. I added a failing test to illustrate this:

~~~ python
    self.assertTrue(S0(['the', 'lazy', 'smelly', 'lazy', 'smelly', 'lazy', 'smelly', 'lazy', 'smelly', 'dog', 'ate', 'noisily', 'EOI']), 'Incorrect')
~~~

FYI, the style I was hoping to see was something like:

~~~ python
import unittest
from functools import reduce
ERROR  = "ERROR"
OK     = "OK"

states = {
  0:  { "the":    1 },
  1:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  2:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  3:  { "ate":    4, "ran":     4 },
  4:  { "slowly": 5, "noisily": 5, "EOI": OK },
  5:  { "EOI":    OK }
}

def getNextState(currentState, nextWord):
    return states.get(currentState,{}).get(nextWord, ERROR)

def stateMachine(sentence):
    return reduce(getNextState, sentence, 0) == OK
~~~